### ISSUE TITLE
Can't load the image on app.vercel.pub

### ISSUE THREAD

**Issue by michalkordyzon** [vercel/platforms#298]
Hello, 
When loading thumbnail image on my site on app.vercel.pub, I get the following message:
```
400: BAD_REQUEST
Code: INVALID_IMAGE_OPTIMIZE_REQUEST
ID: arn1::dkrnn-1692546290759-c5a96c14ad1b  
```

**Comment by 1977744311** [comment]
I've deployed my website on my own Vercel instance, created a blog, and uploaded images, but it seems like the images are also not loading successfully. I'm encountering the same 400 error. How can I fix this?
@steven-tey

**Comment by johnsonchandra** [comment]
you must add in next.config.js your vercel blob url, under images domains
you can see the subdomain in your BLOB_READ_WRITE_TOKEN, in my case e.g.
vercel_blob_rw_s2PJN3q94TmHvmPu_xxxxxxxxxxxx
so you have to add 1 line:

images: {
    domains: [
      ....,
      "s2pjn3q94tmhvmpu.public.blob.vercel-storage.com"
    ],

**Comment by douglasmarianodev** [comment]
BAD REQUEST 400

Request URL:
https://blog.starscode.com.br/_next/image?url=https%3A%2F%2Febk2kj3ob50gly7b.public.blob.vercel-storage.com%2FIgcrwIR-NzHnpVxxwYvQXyWvsni4vI0G5Iozlr.jpeg&w=1200&q=75
Request Method:
GET
Status Code:
400 Bad Request
Remote Address:
76.76.21.164:443
Referrer Policy:
strict-origin-when-cross-origin

bug mesmo adicionando o dominio blob no next.config.js, continua com o bug, ainda não foi resolvido.

**Comment by douglasmarianodev** [comment]
> you must add in next.config.js your vercel blob url, under images domains you can see the subdomain in your BLOB_READ_WRITE_TOKEN, in my case e.g. vercel_blob_rw_s2PJN3q94TmHvmPu_xxxxxxxxxxxx so you have to add 1 line:
> 
> images: { domains: [ ...., "s2pjn3q94tmhvmpu.public.blob.vercel-storage.com" ],

ainda continua com bug, mesmo seguindo esse procedimento.
Tem mais alguma coisa que precisa fazer no código?

**Comment by joshualinog** [comment]
So here is how I resolved it:

# TLDR
**option 1a**-Login to vercel, go to project, click on storage, enter into vercel blob instance 

**option 1b**-or just go to your .env file and copy the 16 or so digits between the `vercel_blog_rw_`**xxxxxxxxxxxxxxxx**`really-long-alpha-numeric-string` in your `BLOB_READ_WRITE_TOKEN` property

now go to your `next.config.js file` and in the images object, inside the `remotePatterns` array, write this object:

## `{ hostname: "xxxxxxxxxxxxxxxx.public.blob.vercel-storage.com" },`

### important: make sure the subdomain you're creating is not pasted like this:
XxXxxxXXXxxxX. IT IS CASE-SENSITIVE, PUT IT ALL IN LOWERCASE.



### More in-depth

1. I had to figure out how to implement solutions from combining answers from three users:

- https://github.com/vercel/platforms/issues/298#issuecomment-1811046449
- https://github.com/vercel/platforms/issues/298#issuecomment-1770498017
- https://github.com/vercel/next.js/discussions/20138#discussioncomment-3073510

3. I then logged into Vercel, clicked on the project instance itself and clicked on the tab that says "storage"
4. I went into the Vercel blob instance and inspected an individual image that was giving the 400 bad request error in my Chrome browser.
5. I right clicked that image and try to open it in a new tab in my Chrome browser (Open image in new tab)
6. Chrome attempted to download the image, but the url was still editable with my mouse: I realized that because the remote patterns object array is case sensitive I had to change my subdomain to all lowercase from mixed casing. This solved it.

**Comment by sajjadhussain14** [comment]
I had the same issue with external images not optimizing in Next.js! After much debugging, here's the solution that worked for me:"

🚨 The Problem
tsx
Copy
// This breaks in production (Vercel)
<Image
  src="https://www.abc.com/prodimages/123.jpg"
  alt="Product"
  width={300}
  height={300}
/>
❌ Error: 400: INVALID_IMAGE_OPTIMIZE_REQUEST
❌ Why: Next.js blocks unconfigured external domains

🛠 The Fix
1️⃣ First, whitelist your domain in next.config.js:

javascript
Copy
module.exports = {
  images: {
    remotePatterns: [
      { 
        protocol: 'https',
        hostname: 'abc.com', // ← Your domain here
      },
    ],
  },
}
2️⃣ Then, add a loader to handle URL formatting:

tsx
Copy
const customLoader = ({ src, width, quality }) => {
  return `${src}?w=${width}&q=${quality || 75}`
}

// Usage:
<Image
  loader={customLoader}
  src="https://www.abc.com/prodimages/123.jpg"
  alt="Product"
  width={300}
  height={300}
/>
✅ Why This Works
remotePatterns tells Next.js which domains to optimize

loader ensures proper URL structure for the image optimizer

Works perfectly on Vercel and other hosts
